pkg/registry/rbac/clusterrole/policybased/storage.go

```diff
@@ -68,8 +80,16 @@ func (s *Storage) Update(ctx genericapirequest.Context, name string, obj rest.Up
 
 		rules := clusterRole.Rules
 		if err := rbacregistryvalidation.ConfirmNoEscalation(ctx, s.ruleResolver, rules); err != nil {
-			return nil, errors.NewForbidden(groupResource, clusterRole.Name, err)
+			return nil, apierrors.NewForbidden(groupResource, clusterRole.Name, err)
+		}
+		// to change the aggregation rule, since it can gather anything and prevent tightening, requires * on *.*
```

Comment:
> Don't you also need to check for the creation or update of a cluster role that will then get aggregated into a different cluster role via an AggregationRule?

When you modify a given ClusterRole, you have to all the permissions you're trying to add.  That means that you have the permission that will eventually be added to the aggregated cluster role and thus added to other users.  We're trying to protect against escalation and I think this does that.  

In the other direction, we have to protect against, "aggregate cluster admin into me".  Forcing a super-user does that.



> The situation I am thinking of is:
> 
> Cluster Role A { aggregates Cluster Role B and Cluster Role C via label D }
> Cluster Role B (has label D) = Rule 1
> Cluster Role C (has label D) = Rule 2
> 
> I only have Rule 2, so I can only pass the escalation check for C. I remove the label from C. This causes A to be tightened (which I normally could not do because I do not have Rule 1). The reverse is also true (if I can create or update a cluster role with label D, I can cause A to be updated even if I would normally not be able to do that).

We're primarily focused on escalation cases. The idea being, "if you have the power to mess with some roles, we need to prevent you from adding more permissions to yourself".  Are you worried about a case where a cluster-admin trusts someone to control what a particular clusterrole can do, but doesn't want that person to be able to control bindings of that role (that's what it sounds like you're describing).  If the role-admin already has the power to perform an action, he could just as easily say "write to this proxy and I'll perform actions for you".  What are you really preventing?

Modifying cluster-roles is already pretty privileged and I'm not seeing how it allows a user to escalate himself (which is what we're trying to protect against).  Given how people would handle this today (secondary binding or direct modification), the power seems the same to me.